controllers/vspherevm_controller.go

```diff
@@ -275,6 +269,15 @@ func (r vmReconciler) Reconcile(ctx goctx.Context, req ctrl.Request) (_ ctrl.Res
 		}
 	}
 
+	clusterModuleInfo, err := r.fetchClusterModuleInfo(vsphereCluster, machine)
```

Comment:
we shouldn't swallow the notfound error in this function, as it makes most the logic bellow unclear. Please file an issue for this.